builtins/amp-ad.js

```diff
@@ -317,36 +320,69 @@ export function installAd(win) {
     /** @override  */
     overflowCallback(overflown, requestedHeight) {
       if (overflown) {
-        const targetOrigin =
-            this.iframe_.src ? parseUrl(this.iframe_.src).origin : '*';
-        postMessage(
-            this.iframe_,
-            'embed-size-denied',
-            {requestedHeight: requestedHeight},
-            targetOrigin,
-            /* opt_is3P */ true);
+        this.onResizeFailed_(requestedHeight);
       }
     }
 
     /**
-     * Updates the elements height to accommodate the iframe's requested height.
-     * @param {number} newHeight
+     * Updates the element's dimensions to accommodate the iframe's
+     *    requested dimensions.
+     * @param {number|undefined} newWidth
+     * @param {number|undefined} newHeight
      * @private
      */
-    updateHeight_(newHeight) {
+    updateDimensions_(newWidth, newHeight) {
+      if (!newWidth && !newHeight) {
+        return;
+      }
+      if (newWidth && typeof newHeight != 'number') {
```

Comment:
Math.max will only guarantee a number when it has a value on data.height  (above),  this is also to prevent NaN s/strings that might happen based on user input.